src/vision/status_quo/barbara_tries_unix_socket.md

```diff
@@ -0,0 +1,314 @@
+
+## The screen stares at Barbara
+
+Barbara wrote some code according to the comments Barbara saw and read some docs like what is `handshake` to roughly know what it does. Barbara compile and it shows a warning, the `connection` variable is not used:
+```
+warning: unused variable: `connection`
+ --> src/main.rs:9:30
+  |
+9 |     let (mut request_sender, connection) = conn::handshake(stream).await?;
+  |                              ^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_connection`
+  |
+  = note: `#[warn(unused_variables)]` on by default
+```
+
+Barbara then runs the program. Barbara was stares at the screen and the screen stares at her. Barbara waited and ... it was stuck. So Barbara decides to look at the logs and run it again with `env RUST_LOG=trace cargo r`, and it was indeed stuck, but not sure where.
+```
+ TRACE mio::poll > registering event source with poller: token=Token(0), interests=READABLE | WRITABLE
+```
+
+Barbara try adding `println!` all over the code but it was still stuck, so Barbara try asking for help. Thanks to the welcoming Rust community, Barbara got help quickly in this case. It seemed like Barbara missed the `connection` which is a culprit and it was in the parent module of the docs Barbara read.
+
+Barbara added the missing piece to `.await` for the `connection`, all the while Barbara thought it will work if it was `.await`-ed but in this case having required to await something else to work is a surprise.
+
+```rust
+    tokio::spawn(async move {
+        if let Err(e) = connection.await {
+            eprintln!("error: {}", e);
+        }
+    })
+    
+    let request = ...
+```
+
+Barbara run the code and it works now, yay! Barbara want to try to reuse the connection to send subsequent HTTP request. Barbara duplicates the last block and it runs.
+
+## Mysterious second request
+
+Some time later, Barbara was told that the program did not work on second request. Barbara tried it but it works. To double confirm, when Barbara tried it again it did not work. Rather than getting stuck, this time there is a error message, which is somewhat better but Barbara did not understand.
+
+The second request is so mysterious, it is like the second request playing hide and seek with Barbara. Sometimes it works and sometimes it does not work.
+
+```rust
+ TRACE mio::poll > registering event source with poller: token=Token(0), interests=READABLE | WRITABLE
+Some(Ok(b"Hello World!"))
+ TRACE want      > signal: Want
+ TRACE mio::poll > deregistering event source from poller
+ TRACE want      > signal: Closed
+Error: hyper::Error(Canceled, "connection was not ready")
+```
+
+As a typical method of solving asynchronous issue. Barbara add prints to every await boundaries in the source code to understand what is going on.
```

Comment:
I like this part (somethings the only thing you know how to do is println debugging).